### PR TITLE
Parsing question params from PDDL Predicate

### PR DESCRIPTION
**Pull request by yala**
This is part of my work on #4 
# Added Functionality

Extract:
    Question Type and Question Params (as describe in #2)
From:
    PDDL Format as described in #3 
# How I Tested This

I added a sample question predicate in `valid_predicates.dictionary`, and stepped through `gdb`
to confirm the `s_QuestionType` and `s_QuestionQuery` had the expected values
# Nitty Gritty Details of How it Works

given a `valid_predicate.dictionary` entry `INDEX|1|question object fruit` (as described in #3 )
The thing returned by `PddlString` is of the form

`"(< (question object fruit ) ) > 0"`

`s_QuestionString` trims this to just `"question object fruit"`
For this example,

`s_QuestionType = "object"`
`s_QuestionQuery = "fruit"`
In another example, `s_QuestionQuery` could equal `"have fruit furnace glass"`, achieving nicola's calling convention

`i_QueryIndex` is just the index in the `s_QuestionString` to start the substring to find the query.

At a high level, this directly interfaces the PDDL semantics with the IR semantics

I don't think descriptions this low level belong in comments, i find them distracting from the point and aren't much simpler than the code itself. 
# Next Steps
- [ ] Interfacing with @nicola's code for answers
- [ ] Handling with question answers
- [ ] Not forwarding questions to MetricFF
- [ ] Making everything configurable

---

Reviewers: @nicola @sebboyer

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -35,3 +35,8 @@ tmp
 
 dev_notes.txt
 **/*.pyc
+=======
+=======
+*.swp
+
+dev_notes.txt
```
Comment (nicola) [inline .gitignore]: r+

File: code/model/SubgoalPolicy.cpp [hunk 2]
```diff
@@ -18,6 +18,7 @@ Subgoal::Subgoal (void)
 	p_PddlSubgoalPredicate = NULL;
 	b_IsLastSubgoalToTarget = false;
 	b_ForcedSequenceEnd = false;
+  b_isQuestion = false;
```
Comment (nicola) [inline code/model/SubgoalPolicy.cpp]: bad indentation

File: code/model/SubgoalPolicy.cpp [hunk 3]
```diff
@@ -1752,6 +1752,26 @@ void SubgoalPolicy::SampleSubgoalSequence (const Problem& _rProblem,
 		pSubgoal->p_PddlSubgoalPredicate
 			= vec_CandidatePredicates [pSubgoal->i_SubgoalSelection];
 
+    //TODO: Add Config Check to make sure this is valid. Else if question found and
+    // config, throw an error
+    if (0 == pSubgoal->p_PddlSubgoalPredicate->s_Name.compare("question")) {
+      pSubgoal->b_isQuestion = true;
+      String_dq_t dq_QuestionArgs;
+      //Parse Question from PddlString
+      String s_PredicateString = pSubgoal->p_PddlSubgoalPredicate->GetPddlString();
```
Comment (nicola) [inline code/model/SubgoalPolicy.cpp]: what does this return? can you give us an example?

File: code/model/SubgoalPolicy.cpp [hunk 4]
```diff
@@ -829,7 +830,8 @@ void SubgoalPolicy::LoadGoldLengthFile(void)
 	ITERATE (String_dq_t, dqLines, iterLine)
 	{
 		// Problem | length
-		String_dq_t dqSplit;
+    String_dq_t dqSplit;
```
Comment (nicola) [inline code/model/SubgoalPolicy.cpp]: indentation + comments

File: code/model/SubgoalPolicy.cpp [hunk 5]
```diff
@@ -1752,6 +1752,26 @@ void SubgoalPolicy::SampleSubgoalSequence (const Problem& _rProblem,
...
+      String_dq_t dq_QuestionArgs;
+      //Parse Question from PddlString
+      String s_PredicateString = pSubgoal->p_PddlSubgoalPredicate->GetPddlString();
+      size_t i_Start = s_PredicateString.rfind("(") + 1;
+      size_t i_End = s_PredicateString.find(")");
+      String s_QuestionString = s_PredicateString.substr(i_Start, i_End - i_Start);
+      s_QuestionString.Split(dq_QuestionArgs, ' ');
+      //Parse Question type and query from question
+      String s_QuestionType = dq_QuestionArgs[1];
+      size_t i_QueryIndex = s_QuestionString.find(dq_QuestionArgs[2]);
```
Comment (nicola) [inline code/model/SubgoalPolicy.cpp]: what does queryIndex means?

File: code/model/SubgoalPolicy.cpp (hunk 2 above)
Comment (yala) [inline code/model/SubgoalPolicy.cpp]: it's weird. My new editor setup keeps messing up my indentation. Even though it looks all equally aligned in my editor..

W.e, i'll reindent through here

File: code/model/SubgoalPolicy.cpp (hunk 5 above)
Comment (yala) [inline code/model/SubgoalPolicy.cpp]: @nicola 

given a `valid_predicate.dictionary` entry `INDEX|1|question object fruit` (as described in #3 )
The thing returned by PddlString is of the form 

`"(< (question object fruit ) ) > 0"`

`s_QuestionString` trims this to just `"question object fruit"` 
For this example,

`s_QuestionType = "object"`
`s_QuestionQuery = "fruit"`
In another example, `s_QuestionQuery` could equal `"have fruit furnace glass"`, achieving your calling convention

`i_QueryIndex` is just the index in the `s_QuestionString` to start the substring to find the query. 

At a high level, this directly interfaces the PDDL semantics with the IR semantics you spec'ed

I don't think descriptions this low level belong in comments, i find them distracting from the point, ie. `Parse Question from PddlString` and `Parse Question type and query from question` and totally redundant.

Comment (nicola) [inline code/model/SubgoalPolicy.cpp]: I think you need to make clear in the code what you mean by query (and also what you have described here). You don't have to comment line by line.

If you don't leave such description me and seb will never understand the hacking behind this, or words like "query" in a "question"

**Pull request merged**